Decouple `logging` from ConfigLoader (#2642)
kedro-org/kedro: RELEASE.md

### CLI
* Removed deprecated `kedro docs` command.

### ConfigLoader

> * `logging` is removed from `ConfigLoader`, it is recommended to use the environment variable`KEDRO_LOGGING_CONFIG`.

Reviewer: ```suggestion
* `logging` is removed from `ConfigLoader` in favour of the environment variable `KEDRO_LOGGING_CONFIG`.
```

### Other
* Removed deprecated `kedro.extras.ColorHandler`.
* The Kedro IPython extension is no longer available as `%load_ext kedro.extras.extensions.ipython`; use `%load_ext kedro.ipython` instead.
* Anonymous nodes are given default names of the form `<function_name>([in1;in2;...]) -> [out1;out2;...]`, with the names of inputs and outputs separated by semicolons.
*

## Migration guide from Kedro 0.18.* to 0.19.*
### DataSets
* If you use `APIDataSet`, move all `requests` specific arguments (e.g. `params`, `headers`), except for `url` and `method`, to under `load_args`.
### Logging

> `logging.yml` is now independent of Kedro's environment and only used if `KEDRO_LOGGING_CONFIG` is set to point to it.

Reviewer: ```suggestion
`logging.yml` is now independent of Kedro's run environment and only used if `KEDRO_LOGGING_CONFIG` is set to point to it.
```
